shinydashboard/lantern/functions/endpoints.R

```diff
@@ -197,4 +200,159 @@ get_avg_response_time <- function(db_connection) {
+    AND smart_http_response=200")) %>%
+    collect() %>%
+    tidyr::replace_na(list(vendor_name = "Unknown")) %>%
+    tidyr::replace_na(list(fhir_version = "Unknown"))
+}
+
+# Summarize the count of capabilities reported in SMART Core Capabilities JSON doc
+get_smart_response_capability_count <- function(endpoints_tbl) {
+  res <- endpoints_tbl %>% 
+    group_by(fhir_version,capability) %>%
+    count() %>%
+    rename("FHIR Version" = fhir_version, Capability = capability, Endpoints = n)
+  res
+}
+
+# Query fhir endpoints and return list of endpoints that have 
+# returned a valid JSON document at /.well-known/smart-configuration
+# This implies a smart_http_response of 200.
+# 
+get_well_known_endpoints_tbl <- function(db_connection) {
+  res <- tbl(db_connection,
+    sql("SELECT e.url, e.organization_names, v.name as vendor_name,
+      f.capability_statement->>'fhirVersion' as fhir_version
+    FROM fhir_endpoints_info f 
+    LEFT JOIN vendors v on f.vendor_id = v.id
+    LEFT JOIN fhir_endpoints e
+    ON f.id = e.id
+    WHERE f.smart_http_response = 200
+    AND jsonb_typeof(f.smart_response) = 'object'")) %>%
+    collect() %>%
+    tidyr::replace_na(list(vendor_name = "Unknown")) %>%
+    tidyr::replace_na(list(fhir_version = "Unknown"))
+}
+
+# get count of well known endpoints returning http 200 (but not
+# checking if valid SMART core capability doc returned)
+get_well_known_endpoints_count <- function(db_connection) {
+  res <- tbl(db_connection,
+      sql("SELECT count(*) from fhir_endpoints_info
+          WHERE smart_http_response = 200")) %>% collect() %>% pull(count)
+  as.integer(res)
+}
+
+# Find any endpoints which have returned a smart_http_response of 200
+# at the well known endpoint url /.well-known/smart-configuration
+# but did NOT return a valid JSON document when queried
+get_well_known_endpoints_no_doc <- function(db_connection) {
+  res <- tbl(db_connection,
+    sql("SELECT f.id, e.url, f.vendor_id, e.organization_names, v.name as vendor_name,
+      f.capability_statement->>'fhirVersion' as fhir_version,
+	    f.smart_http_response,
+	    f.smart_response
+    FROM fhir_endpoints_info f 
+    LEFT JOIN vendors v on f.vendor_id = v.id
+    LEFT JOIN fhir_endpoints e
+    ON f.id = e.id
+    WHERE f.smart_http_response = 200
+    AND jsonb_typeof(f.smart_response) <> 'object'")) %>%
+    collect() %>%
+    tidyr::replace_na(list(vendor_name = "Unknown")) %>%
+    tidyr::replace_na(list(fhir_version = "Unknown"))
+}
+
+# Return a summary table of information about endpoint security statements
+get_well_known_endpoint_counts <- function(db_connection) {
+  res <- tribble(
+    ~Status, ~Endpoints,
+    "Total Indexed Endpoints", as.integer(app_data$fhir_endpoint_totals$all_endpoints),
+    "Endpoints with successful response (HTTP 200)", as.integer(app_data$response_tally$http_200),
+    "Well Known URI Endpoints with succesful response", get_well_known_endpoints_count(db_connection),
```

Comment:
Should also specify (HTTP 200) int he same way that was done for successful endpoint response in the line above

Reply:
Will update table label to match